### ISSUE TITLE
Dia isn't generating anything

### ISSUE THREAD

**Issue by znraznra** [nari-labs/dia#154]
Cloned repo and followed all instructions in the readme. Ran CMD as administrator so that symlinks can be used. Got 1 warning 

`C:\Users\arnzl\AppData\Local\Programs\Python\Python312\Lib\site-packages\torch\nn\utils\weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)`

Gradio loaded immediately and I can access the site through localhost:7860. When I try generating an audio, nothing happens as if it's not registering anything. I tried switching between browsers (Firefox, Edge, Brave) and the issues still persist.

Here's my system specifications:

![Image](https://github.com/user-attachments/assets/5ef3da61-a1de-4bdd-b7b2-b2cbb2c859dc)

I'm not sure what's wrong with this. I've been trying to refresh while trying to generate and nothing ever changes in the cmd screen.

![Image](https://github.com/user-attachments/assets/f300a3f8-ef5f-489c-a4d6-5b9587b491a0)

Are there some prerequisites that I need to look for? What's wrong with this exactly?

**Comment by RobertAgee** [comment]
you're generating on cpu instead of cuda

**Comment by znraznra** [comment]
> You're generating on cpu instead of cuda. I wouldn't recommend running this unless you have GPU.

Hi, do you have any idea how to set it to GPU? I checked the readme and it doesn't say how.

**Comment by RobertAgee** [comment]
cuda support on windows is notoriously bad. What kind of GPU and how much vram do you have?

in my case, on my computer I need to run this in Linux on WSL2 for it to detect my GPU correctly. it's not a matter of the repo or a setting within it, it's just the cuda packages in a windows environment as far as I can tell

**Comment by znraznra** [comment]
> cuda support on windows is notoriously bad. What kind of GPU and how much vram do you have?
> 
> in my case, on my computer I need to run this in Linux on WSL2 for it to detect my GPU correctly. it's not a matter of the repo or a setting within it, it's just the cuda packages in a windows environment as far as I can tell

I do have an RTX 2070 Super 8GB. I know that these tools usually favor 30-series but yeah, just trying my luck.

**Comment by RobertAgee** [comment]
Ok, that's good to know! I also have a potato 8gb card and i needed to do some optimizations to run faster and within memory (chunking the input). #PR159 has the code if you want to try that.

**Comment by znraznra** [comment]
> Ok, that's good to know! I also have a potato 8gb card and i needed to do some optimizations to run faster and within memory (chunking the input). #PR159 has the code if you want to try that.

Glad to know that, thank you for the tip! I'll give an update if it works. Thank you again.

**Comment by znraznra** [comment]
I reinstalled my Dia and used a conda environment for that, then tried the optimization fixes. It's perfectly working. Thank you @RobertAgee

**Comment by wzkris** [comment]
> I reinstalled my Dia and used a conda environment for that, then tried the optimization fixes. It's perfectly working. Thank you [@RobertAgee](https://github.com/RobertAgee)

Hi. Dose Dia support NVIDIA GeForce GT 730? I found that use CPU running and dosen't generate anything.

![Image](https://github.com/user-attachments/assets/4a29e95e-bfc5-4f4d-b9ae-0b907e5202ab)

**Comment by znraznra** [comment]
> > I reinstalled my Dia and used a conda environment for that, then tried the optimization fixes. It's perfectly working. Thank you [@RobertAgee](https://github.com/RobertAgee)
> 
> Hi. Dose Dia support NVIDIA GeForce GT 730? I found that use CPU running and dosen't generate anything.
> 
> ![Image](https://github.com/user-attachments/assets/4a29e95e-bfc5-4f4d-b9ae-0b907e5202ab)

You need the RTX cards' tensor cores to make Dia use the GPU. Other than that, it'll default to CPU.

**Comment by buttercrab** [comment]
Closing resolved issue.
